### DISCUSSION TITLE
Explain how to get points from a mesh

### DISCUSSION THREAD

**Discussion by oleg-alexandrov** [CGAL/cgal#7341] [General]
The documentation at https://doc.cgal.org/latest/Surface_mesh/index.html focuses on iterators and circulators. It would be nice to have an example to to actually get a vertex value from a mesh. This looks like a very primitive operation to me. 

It may also help to show to create a list of points as well, or a point set, given that many operations in CGAL (such as Poisson reconstruction) start with vertices, and not with meshes, even if the goal is mesh processing.

**Comment by sloriot** [comment]
I guess you mean accessing the coordinates of a vertex. It is a bit burred in the doc [here](https://doc.cgal.org/latest/Surface_mesh/index.html#sectionSurfaceMesh_properties): *One property is maintained by default, namely "v:point". The value of this property has to be supplied when adding a new point to the data structure via [Surface_mesh::add_vertex()](https://doc.cgal.org/latest/Surface_mesh/classCGAL_1_1Surface__mesh.html#a6f899386b0667edc64cfae79cc93386e). The property can be directly accessed using [Surface_mesh::points()](https://doc.cgal.org/latest/Surface_mesh/classCGAL_1_1Surface__mesh.html#a0887ce92d0d65bcd1128c313bc9d2d07) or [Surface_mesh::point(Surface_mesh::Vertex_index v)](https://doc.cgal.org/latest/Surface_mesh/classCGAL_1_1Surface__mesh.html#aaf6c8f0ab1ccc8a7b8b07de618aca2f8).*  [This examples](https://doc.cgal.org/latest/Surface_mesh/Surface_mesh_2sm_properties_8cpp-example.html) uses the `point()` method.

A list of points is just a range of points like `std::vector<Point_3>`.

**Reply by oleg-alexandrov** [reply]
Thank you. That is helpful. It is still easier for the user if a little spelled out example exists saying, here's a mesh, here how one iterates over it, and here's how coordinates of vertices are printed. Likely it could be in Example 3.1, which prints the vertex descriptor, so could as well print the vertex value.

**Reply by oleg-alexandrov** [reply]
Here's what I have in mind. Maybe it does not align with your philosophy but gives an idea of what may help. https://github.com/CGAL/cgal/pull/7342

**Comment by oleg-alexandrov** [comment]
Fixed by https://github.com/CGAL/cgal/pull/7343 and https://github.com/CGAL/cgal/pull/7342.
